### ISSUE TITLE
[Question]: not defined props

### ISSUE THREAD

**Issue by CosticaPuntaru** [Futhark/ngx-observable-input#2]
the fallowing code fails
```ts
  @ObservableInput()
  @Input('productId')
  public productId$: Observable<string>
  public ngOnInit(): void {
   this.productDetails$ = this.productId$.pipe(switchMap( (productId) => this.service.getProduct(productId) ));
}
```

1. should the observable be initiated even when the component is used whiteout the given prop?

2. should i define the prop as:

```ts
  public productId$?: Observable<string>
```
3.  dose this work as expected?

```
public productId$: Observable<string> = new ReplySubject(1)
```

**Comment by Futhark** [comment]
I think that It should be a stream with "undefined" as first item, or maybe better, the decorator should accept default value as a parameter. Either way, current behavior is in fact an issue. We should be safely use that stream even if it is not used. I would try to fix that but it will be released as new major version because it breaks compatibility with the previous one (it is possible that someone  used that behavior in it's app logic).

**Comment by Futhark** [comment]
@CosticaPuntaru I've released the version 3.0.0 and updated README.md. Please check the latest build.
